Tests/SonarScanner.MSBuild.PreProcessor.Test/EmbeddedAnalyzerInstallerTests.cs

```diff
@@ -201,23 +201,25 @@ public void EmbeddedInstall_CachingScenarios()
 
         // 1. Empty cache -> cache miss -> server called
         var actualFiles = testSubject.InstallAssemblies([requestA]);
-        server.ReceivedWithAnyArgs(1).TryDownloadEmbeddedFile(null, null, null);    // should have tried to download
+        server.Received(1).TryDownloadEmbeddedFile("p111", "p1.zip", Arg.Any<string>());
+        server.ReceivedWithAnyArgs(1).TryDownloadEmbeddedFile(null, null, null);    // no other downloads
 
         AssertExpectedFilesReturned(expectedPlugin111Paths, actualFiles);
         AssertExpectedFilesExist(expectedPlugin111Paths);
         AssertExpectedFilesInCache(4, localCacheDir); // only files for the first request should exist
 
         // 2. New request + request -> partial cache miss -> server called only for the new request
         actualFiles = testSubject.InstallAssemblies([requestA, requestB]);
-        server.ReceivedWithAnyArgs(2).TryDownloadEmbeddedFile(null, null, null);    // new request
+        server.Received(1).TryDownloadEmbeddedFile("p222", "p2.zip", Arg.Any<string>());
+        server.ReceivedWithAnyArgs(2).TryDownloadEmbeddedFile(null, null, null);    // 1 from before + 1 new
```

Comment:
I think it is better to `ClearReceivedCalls()` and assert `DidNotReceivedWithAnyArgs()`
https://nsubstitute.github.io/help/clear-received-calls/index.html
It makes the UT clearer and the comment can be removed.